packages/hash/frontend/src/lib/subgraph.ts

```diff
@@ -0,0 +1,231 @@
+import {
+  Vertex,
+  DataTypeVertex,
+  PropertyTypeVertex,
+  LinkTypeVertex,
+  EntityTypeVertex,
+  EntityVertex,
+  LinkVertex,
+  PersistedDataType,
+  PersistedPropertyType,
+  PersistedLinkType,
+  PersistedEntityType,
+} from "@hashintel/hash-shared/graphql/types";
+import { BaseUri } from "@blockprotocol/type-system-web";
+
+import { Subgraph } from "../graphql/apiTypes.gen";
+import { EntityResponse } from "../components/hooks/blockProtocolFunctions/knowledge/knowledge-shim";
+
+// ------------------- Type Guards to use inside .filters -------------------
+
+export const isDataTypeVertex = (vertex: Vertex): vertex is DataTypeVertex => {
```

Comment:
These are necessary because Typescript doesn't seem to understand that `vertex.kind === "dataType"` constrains the type if you continue past a `filter` (for example with a `map`)